lib/service/api/nfl_athlete_api.dart

```diff
@@ -1,62 +1,171 @@
+import 'dart:convert';
+
 import 'package:ax_dapp/app/config/app_config.dart';
 import 'package:ax_dapp/service/api/models/player_ids.dart';
 import 'package:ax_dapp/service/athlete_models/athlete_price_record.dart';
 import 'package:ax_dapp/service/athlete_models/nfl/nfl_athlete.dart';
 import 'package:ax_dapp/service/athlete_models/nfl/nfl_athlete_stats.dart';
+import 'package:ax_dapp/service/athlete_models/price_record.dart';
 import 'package:dio/dio.dart';
-import 'package:retrofit/http.dart';
+import 'package:http/http.dart' as http;
+
+class NFLAthleteAPI {
+  NFLAthleteAPI(this.dio) {
+    initState();
+  }
+
+  final Dio dio;
+  late String baseDataUrl;
+  static const DEFAULT_FROM = '2023-01-01';
+  static const DEFAULT_UNTIL = '2023-12-31';
+  static const DEFAULT_TIME = 'Day';
+
+  Future<void> initState() async {
+    const cidUrl =
+        'https://raw.githubusercontent.com/AthleteX-DAO/sports-cids/main/nfl.json';
+    final response = await http.get(Uri.parse(cidUrl));
+    if (response.statusCode == 200) {
+      final decodedData = json.decode(response.body);
+      final cid = await decodedData['directory'] as String;
+      baseDataUrl = 'https://$cid.ipfs.nftstorage.link';
+    } else {
+      throw Exception('Failed to fetch data');
+    }
+  }
+
+  Future<List<NFLAthlete>> getAllPlayers() async {
+    final athletes = <NFLAthlete>[];
+    final url = '$baseDataUrl/ALL_PLAYERS';
+    final jsonString = await dio.get(url);
```

Comment:
Also on line 39 inside getAllPlayers dio.get is a type <T> but since it does not know the type it defaults to dynamic. I think it would be better to work with known data types rather than dynamic to prevent errors from showing up in our app.